### PR TITLE
chore: parameterize price feed replacement (check liquidation)

### PR DESCRIPTION
**Pull request by dckc**
DRAFT until:
 - [ ] fill in the right per-brand proposals
 - [x] adjust `setup`, `outcome` constants

stacked on
 - #9999

was:
 - #9748

## Description
```js
/**
 * @file  The goal of this test is  to see that the
 * upgrade scripts re-wire all the contracts so new auctions and
 * price feeds are connected to vaults correctly.
 *
 * 1. enter a bid
 * 2. force prices to drop so a vault liquidates
 * 3. verify that the bidder gets the liquidated assets.
 */
```

### CODE REVIEW

**Comment by dckc** [comment]
runs like this so far:

```
...
  ─

  1. setupVaults; placeBids

  Error: Test finished without running any assertions



  run replace-price-feeds proposals
  Error: Test finished without running any assertions



  2. trigger liquidation by changing price

  test/bootstrapTests/price-feed-replace.test.ts:97

   96:   await advanceTimeTo(NonNullish(liveSchedule.nextDescendingStepTime));
   97:   t.like(readLatest(metricsPath), {                                    
   98:     numActiveVaults: 0,                                                

  Difference (- actual, + expected):

    {
      liquidatingCollateral: {
  -     value: 0n,
  +     value: 45000000n,
      },
      liquidatingDebt: {
  -     value: 0n,
  +     value: 309540000n,
      },
      lockedQuote: null,
  -   numActiveVaults: 1,
  +   numActiveVaults: 0,
  -   numLiquidatingVaults: 0,
  +   numLiquidatingVaults: 1,
    }

  › Object.value [as like] (file:///home/connolly/projects/agoric-sdk/node_modules/@endo/ses-ava/src/ses-ava-test.js:164:35)
  › <anonymous> (test/bootstrapTests/price-feed-replace.test.ts:97:5)



  3. verify liquidation
  test/bootstrapTests/price-feed-replace.test.ts:113

   112:   await advanceTimeBy(3, 'minutes');                                       
   113:   t.like(readLatest(`published.auction.book${managerIndex}`), {            
   114:     collateralAvailable: { value: scale6(setup.auction.start.collateral) },

  Difference (- actual, + expected):

    {
      collateralAvailable: {
  -     value: 0n,
  +     value: 45000000n,
      },
      startCollateral: {
  -     value: 0n,
  +     value: 45000000n,
      },
  -   startProceedsGoal: null,
  +   startProceedsGoal: {
  +     value: 309540000n,
  +   },
    }

  › Object.value [as like] (file:///home/connolly/projects/agoric-sdk/node_modules/@endo/ses-ava/src/ses-ava-test.js:164:35)
  › <anonymous> (test/bootstrapTests/price-feed-replace.test.ts:113:5)

  ─

  4 tests failed
error Command failed with exit code 1.
```

**Comment by dckc** [comment]
@Chris-Hibbert help me understand why liquidation is not triggering?

_p.s. because the updated price is coming from a feed that vaults/auctions have yet to be introduced to._

<details><summary>✘ [fail]: 3. verify liquidation</summary>

```console
$ yarn test test/bootstrapTests/price-feed-replace.test.ts
...

  ✘ [fail]: 3. verify liquidation
  ─

  3. verify liquidation

  test/bootstrapTests/price-feed-replace.test.ts:125

   124:   // vaultFactory sent collateral for liquidation
   125:   t.like(readLatest(metricsPath), {              
   126:     numActiveVaults: 0,                          

  Difference (- actual, + expected):

    {
      liquidatingCollateral: {
  -     value: 0n,
  +     value: 15000000n,
      },
      liquidatingDebt: {
  -     value: 0n,
  +     value: 100500000n,
      },
      lockedQuote: null,
  -   numActiveVaults: 1,
  +   numActiveVaults: 0,
  -   numLiquidatingVaults: 0,
  +   numLiquidatingVaults: 1,
    }

  › Object.value [as like] (file:///home/connolly/projects/agoric-sdk/node_modules/@endo/ses-ava/src/ses-ava-test.js:164:35)
  › <anonymous> (test/bootstrapTests/price-feed-replace.test.ts:125:5)

  ─

  1 test failed
error Command failed with exit code 1.
```
</details> 


I rebased it on #9999. There's more to do, but I'd like to get this part working.

**Comment by dckc** [comment]
@Chris-Hibbert over to you as of 3d6c4e7

File: packages/boot/test/bootstrapTests/price-feed-replace.test.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  await evalProposal(combined);
+
+  refreshAgoricNamesRemotes();
+  const instancePost = agoricNamesRemotes.instance['ATOM-USD price feed'];
+  t.not(instancePre, instancePost);
+
+  await priceFeedDrivers[collateralBrandKey].refreshInvitations();
+});
+
+test.serial('1. place bid', async t => {
```
Comment (Chris-Hibbert) [inline packages/boot/test/bootstrapTests/price-feed-replace.test.ts:91]: @turadg has pointed out that `test.serial()` doesn't mean "must run in this order". It apparently means "must run one at a time in some order".

Should we do something else for tests that must run in a particular order?

File: packages/inter-protocol/src/proposals/deploy-price-feeds.js [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+  },
+  { name, decimalPlaces },
+) => {
+  const brand = E(agoricNames).provideInertBrand(name, {
+    assetKind: 'nat',
+    decimalPlaces,
+  });
+
+  oracleBrandProduce[name].reset();
+  oracleBrandProduce[name].resolve(brand);
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/proposals/deploy-price-feeds.js:82]: Why is it okay for this to overwrite existing brands? Do we expect to use that ability?

File: packages/inter-protocol/src/proposals/deploy-price-feeds.js [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        namesByAddressAdmin: t,
+        priceAuthority: t,
+        priceAuthorityAdmin: t,
+        startGovernedUpgradable: t,
+        zoe: t,
+      },
+      instance: {
+        produce: t,
+      },
+    },
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/proposals/deploy-price-feeds.js]: installPriceAggregator needs

```
    installation: {
      produce: { priceAggregator },
    },
```

**Review by Chris-Hibbert — COMMENTED** [review]
I have two questions.

I saw that you addressed the missing `consume` power.

File: packages/inter-protocol/src/proposals/deploy-price-feeds.js (hunk 3 above)
Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/deploy-price-feeds.js]: I see that a later commit handled this.

File: packages/boot/test/bootstrapTests/price-feed-replace.test.ts (hunk 1 above)
Comment (turadg) [reply packages/boot/test/bootstrapTests/price-feed-replace.test.ts:91]: when tests must run in a particular order, they're better thought of as one test. if you want more structure around the steps, a little helper can be made to log the progress.

File: packages/inter-protocol/src/proposals/deploy-price-feeds.js (hunk 2 above)
Comment (dckc) [reply packages/inter-protocol/src/proposals/deploy-price-feeds.js:82]: Probably not.

`provideInertBrand(name)` gives the same result for the same name, so it shouldn't make any difference.

But the `.reset()` is probably a distraction.

Comment (Chris-Hibbert) [reply packages/inter-protocol/src/proposals/deploy-price-feeds.js:82]: I'll drop the reset(). Thanks.

**Comment by Chris-Hibbert** [comment]
This has been incorporated into #10074.

**Pull request closed without merging**